modules/runtime/nuxeo-runtime-metrics/src/main/java/org/nuxeo/runtime/metrics/reporter/PrometheusReporter.java

```diff
@@ -69,7 +71,7 @@ public void start(MetricRegistry registry, MetricFilter filter, Set<MetricAttrib
         try {
             server = new HTTPServer(port, true);
         } catch (IOException e) {
-            throw new RuntimeException("Cannot start Prometheus on port " + port, e);
+            throw new IllegalArgumentException("Cannot start Prometheus on port " + port, e);
```

Comment:
`IllegalStateException` maybe? or the `IOException` is always linked to the `port` param?

Reply:
Yes, port conflict is the main cause of IOException here.